### ISSUE TITLE
Allow closing `WSMessageHandler` stream with a close code

### ISSUE THREAD

**Issue by liautaud** [swhitty/FlyingFox#137]
Hi! First of all thanks for the library, it's been really great to use.

Would you be open to adding a way for a `WSMessageHandler` to close the server stream with `.close(code: closeCode)` frames in some way or another? The typical use case is closing a WebSocket connection after some unrecoverable error while wanting the client-side to know what the error actually is–in JavaScript, for instance, you can read `event.code` on a WebSocket `"close"` event.

I could already implement this behavior by falling back to the more cumbersome `WSHandler` level, but it seems like a frequent enough use case that it would make sense at the `WSMessageHandler` level too.

**Comment by swhitty** [comment]
Yeah I think it makes sense.  Just looking at [Mozilla docs](https://developer.mozilla.org/en-US/docs/Web/API/WebSocket/close), would it make sense for something like this?

```swift
public enum WSMessage: @unchecked Sendable, Hashable {
     case text(String)
     case data(Data)
     case close(code: Int = 1000, reason: String? = nil)
}
```

And then I think the `AsyncStream<WSMessage>` (from  `WSMessageHandler`) would then always emit the `.close` message before the sequence ends?

This would also allow servers to send `case .close` to clients.  We could validate that close is only ever sent to clients once..

What do you think?

**Comment by swhitty** [comment]
FYI the main branch has this new `case` allowing servers to send and receive custom close codes and reasons from clients.

I'm interested if this works for your use case.

**Comment by swhitty** [comment]
Available in the [0.22.0](https://github.com/swhitty/FlyingFox/releases/tag/0.22.0) release
